server/api/api/utils.py

```diff
@@ -358,16 +358,19 @@ def unmask_notification_params_secret(
             "Not running in k8s environment, cannot load notification params secret"
         )
 
-    notification_object.secret_params = json.loads(
-        server.api.crud.Secrets().get_project_secret(
-            project,
-            mlrun.common.schemas.SecretProviderName.kubernetes,
-            secret_key=params_secret,
-            allow_internal_secrets=True,
-            allow_secrets_from_k8s=True,
-        )
+    secret = server.api.crud.Secrets().get_project_secret(
+        project,
+        mlrun.common.schemas.SecretProviderName.kubernetes,
+        secret_key=params_secret,
+        allow_internal_secrets=True,
+        allow_secrets_from_k8s=True,
     )
 
+    if secret is None:
+        raise mlrun.errors.MLRunPreconditionFailedError("Provided secret not found")
```

Comment:
```suggestion
        raise mlrun.errors.MLRunPreconditionFailedError(f"Provided secret not found: {params_secret}")
```

Reply:
not sure this is a good idea from security POV

Reply:
If that's a concern then we might not want an indicative error message at all. All the user needs to do is remember their input to gain info from the message without the specific secret name. So maybe just return an empty BadRequest exception?

Reply:
It doesn't always happen as a result of specific user request, since it fires when we send the notification. So, perhaps same PreconditionFailes, but empty